aten/src/ATen/native/BlasKernel.cpp

```diff
@@ -196,15 +198,33 @@ void gemv(char trans, int64_t m, int64_t n, scalar_t alpha, scalar_t *a, int64_t
   } else {
     if (beta != scalar_t(1) && beta != scalar_t(0)) scal<scalar_t>(m, beta, y, incy);
 
+    bool is_low_precision = !std::is_same<opmath_t, scalar_t>::value;
+    std::vector<opmath_t> sum;
+    if (is_low_precision) {
+      sum.resize(m);
+    }
     for (const auto j : c10::irange(n)) {
       scalar_t *column_ = a + lda * j;
       scalar_t z = alpha * x[j * incx];
       for (const auto i : c10::irange(m)) {
         //output values are ignored if beta is 0, and set to 0, nans and infs are not propagated
         if (j==0 && beta==scalar_t(0)) {
-         y[i * incy] = scalar_t(0);
+          if (is_low_precision) {
+            sum[i] = 0;
+          } else {
+            y[i * incy] = 0;
+          }
+        }
+        if (is_low_precision) {
+          sum[i] += z * column_[i];
+        } else {
+          y[i * incy] += z * column_[i];
         }
-        y[i * incy] += z * column_[i];
+      }
+    }
+    if (is_low_precision) {
+      for (const auto i : c10::irange(m)) {
+        y[i * incy] = sum[i];
       }
```

Comment:
You forgot to include the y value in the sum. I'm more than a little concerned this wasn't picked up in CI. Is this code not tested?

```suggestion
      if (beta == scalar_t(0) {
        for (const auto i : c10::irange(m)) {
          y[i * incy] = sum[i];
        }
      } else {
        for (const auto i : c10::irange(m)) {
          y[i * incy] += sum[i];
        }
      }
```

Reply:
Yes you are correct. The test_addmv do not cover this reference path, I added ut to cover it.